posthog/hogql/printer.py

```diff
@@ -351,121 +349,93 @@ def visit_field_symbol(self, symbol: ast.FieldSymbol):
                     return self.visit(ast.AsteriskSymbol(table=ast.TableSymbol(table=resolved_field)))
                 else:
                     return self.visit(
                         ast.AsteriskSymbol(
                             table=ast.TableAliasSymbol(
-                                table=ast.TableSymbol(table=resolved_field), name=symbol.table.name
+                                table_symbol=ast.TableSymbol(table=resolved_field), name=symbol.table.name
                             )
                         )
                     )
 
             field_sql = self._print_identifier(resolved_field.name)
 
-            # :KLUDGE: Legacy person properties handling. Assume we're in a context where the tables have been joined,
-            # and this "person_props" alias is accessible to us.
-            if resolved_field == database.events.person.properties:
-                if not self.context.using_person_on_events:
-                    field_sql = "person_props"
-
             # If the field is called on a table that has an alias, prepend the table alias.
             # If there's another field with the same name in the scope that's not this, prepend the full table name.
             # Note: we don't prepend a table name for the special "person" fields.
-            elif isinstance(symbol.table, ast.TableAliasSymbol) or symbol_with_name_in_scope != symbol:
+            if isinstance(symbol.table, ast.TableAliasSymbol) or symbol_with_name_in_scope != symbol:
                 field_sql = f"{self.visit(symbol.table)}.{field_sql}"
 
-            # TODO: refactor this legacy logging
-            if symbol.name != "properties":
-                real_table = symbol.table
-                while isinstance(real_table, ast.TableAliasSymbol):
-                    real_table = real_table.table
-
-                access_table = (
-                    cast(Literal["event"], "event")
-                    if real_table.table == database.events
-                    else cast(Literal["person"], "person")
-                )
-                self.context.field_access_logs.append(
-                    HogQLFieldAccess(
-                        ["person", symbol.name] if access_table == "person" else [symbol.name],
-                        access_table,
-                        symbol.name,
-                        field_sql,
-                    )
-                )
-
         elif isinstance(symbol.table, ast.SelectQuerySymbol) or isinstance(symbol.table, ast.SelectQueryAliasSymbol):
             field_sql = self._print_identifier(symbol.name)
             if isinstance(symbol.table, ast.SelectQueryAliasSymbol) or symbol_with_name_in_scope != symbol:
                 field_sql = f"{self.visit(symbol.table)}.{field_sql}"
 
+            # :KLUDGE: Legacy person properties handling. Only used within non-HogQL queries, such as insights.
+            if self.context.legacy_person_property_handling and field_sql == "events__pdi__person.properties":
```

Comment:
something like `within_non_hogql_query` is shorter than `legacy_person_property_handling` and carries more information. Then the comment can be deleted

Reply:
This is a much better name 👍